File: src/main/java/com/snow/BuggyJavaCode.py
import os
import sys
import random
import hashlib
import sqlite3

# --- Hardcoded Credentials (Security Issue) ---
USERNAME = "admin"
PASSWORD = "password123"  # Hardcoded password

# --- Unused Variables (Code Smell) ---
unused_var1 = 42
unused_var2 = "I am never used"

# --- Duplicate Code Example (Code Smell) ---
def duplicate_function_1():
    x = 10
    y = 20
    return x + y

def duplicate_function_2():
    x = 10
    y = 20
    return x + y

# --- SQL Injection (Security Issue) ---
def unsafe_query(user_input):
    conn = sqlite3.connect("database.db")
    cursor = conn.cursor()
    query = "SELECT * FROM users WHERE name = '" + user_input + "'"  # Vulnerable to SQL Injection
    cursor.execute(query)
    return cursor.fetchall()

# --- Infinite Loop (Bug) ---
def infinite_loop():
    while True:
        pass  # Never terminates

# --- Mutable Default Argument (Bug) ---
def mutable_arg_bug(data=[]):
    data.append(1)
    return data

# --- Uncaught Exception (Bug) ---
def exception_bug():
    try:
        return 1 / 0  # Division by zero
    except ValueError:  # Catches wrong exception type
        print("ValueError occurred!")

# --- Unused Import (Code Smell) ---
import json

# --- Use of eval() (Security Issue) ---
def eval_security_issue(user_input):
    return eval(user_input)  # Dangerous function

# --- Logging Sensitive Information (Security Issue) ---
def log_sensitive():
    print(f"Logging password: {PASSWORD}")  # Exposing sensitive data in logs

# --- Large Cyclomatic Complexity (Code Smell) ---
def complex_function(x):
    if x > 0:
        if x % 2 == 0:
            if x % 3 == 0:
                if x % 5 == 0:
                    print("Multiple conditions met")

# --- Insecure Randomness (Security Issue) ---
def insecure_random():
    return random.randint(0, 100)  # Should use `secrets` for security-sensitive operations

# --- Dead Code (Code Smell) ---
def dead_code():
    return
    print("This line will never execute")  # Dead code

# --- Memory Leak (Bug) ---
def memory_leak():
    large_list = []
    while True:
        large_list.append("leak")  # Unbounded memory growth

# --- Main Execution ---
if __name__ == "__main__":
    print("Executing buggy script...")
    exception_bug()
    unsafe_query("admin' OR '1'='1")  # SQL Injection Example
    eval_security_issue("__import__('os').system('ls')")  # Dangerous eval
    infinite_loop()  # Causes program to hang

File: src/main/java/com/snow/BuggyJavaCode.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.Statement;
import java.util.Random;
import java.util.ArrayList;

public class BuggyJavaCode {
    
    // --- Hardcoded Credentials (Security Issue) ---
    private static final String USERNAME = "admin";
    private static final String PASSWORD = "password123";  // Hardcoded password

    // --- Unused Variables (Code Smell) ---
    private static int unusedVar1 = 42;
    private static String unusedVar2 = "I am never used";

    // --- Duplicate Code Example (Code Smell) ---
    public static int duplicateFunction1() {
        int x = 10, y = 20;
        return x + y;
    }

    public static int duplicateFunction2() {
        int x = 10, y = 20;
        return x + y;
    }

    // --- SQL Injection (Security Issue) ---
    public static void unsafeQuery(String userInput) {
        try {
            Connection conn = DriverManager.getConnection("jdbc:sqlite:database.db");
            Statement stmt = conn.createStatement();
            String query = "SELECT * FROM users WHERE name = '" + userInput + "'"; // SQL Injection risk
            ResultSet rs = stmt.executeQuery(query);
            while (rs.next()) {
                System.out.println("User: " + rs.getString("name"));
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    // --- Infinite Loop (Bug) ---
    public static void infiniteLoop() {
        while (true) {
            // Never terminates
        }
    }

    // --- Mutable Static Variable (Bug) ---
    private static ArrayList<String> mutableList = new ArrayList<>();

    public static void mutableBug() {
        mutableList.add("Buggy");
    }

    // --- Uncaught Exception (Bug) ---
    public static void exceptionBug() {
        try {
            int result = 1 / 0;  // Division by zero
        } catch (NullPointerException e) {  // Catching wrong exception type
            System.out.println("NullPointerException occurred!");
        }
    }

    // --- Use of eval() equivalent (Security Issue) ---
    public static void evalSecurityIssue(String command) {
        try {
            Runtime.getRuntime().exec(command); // Dangerous execution
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    // --- Logging Sensitive Information (Security Issue) ---
    public static void logSensitive() {
        System.out.println("Logging password: " + PASSWORD); // Exposing sensitive data
    }

    // --- Large Cyclomatic Complexity (Code Smell) ---
    public static void complexFunction(int x) {
        if (x > 0) {
            if (x % 2 == 0) {
                if (x % 3 == 0) {
                    if (x % 5 == 0) {
                        System.out.println("Multiple conditions met");
                    }
                }
            }
        }
    }

    // --- Insecure Randomness (Security Issue) ---
    public static int insecureRandom() {
        Random random = new Random();
        return random.nextInt(100);  // Should use SecureRandom for security-sensitive operations
    }

    // --- Dead Code (Code Smell) ---
    public static void deadCode() {
        return;
        // This line will never execute
        System.out.println("Dead code");
    }

    // --- Memory Leak (Bug) ---
    public static void memoryLeak() {
        ArrayList<String> largeList = new ArrayList<>();
        while (true) {
            largeList.add("leak"); // Unbounded memory growth
        }
    }

    public static void main(String[] args) {
        System.out.println("Executing buggy Java code...");
        exceptionBug();
        unsafeQuery("admin' OR '1'='1");  // SQL Injection Example
        evalSecurityIssue("ls");  // Dangerous command execution
        infiniteLoop();  // Causes program to hang
    }
}
